angularSTEP/src/app/branching-recipe-form/branching-recipe-form.component.ts

```diff
@@ -0,0 +1,253 @@
+import {Component, OnInit} from '@angular/core';
+import {AngularFireAuth} from '@angular/fire/auth';
+import {AngularFirestore} from '@angular/fire/firestore';
+import {ActivatedRoute, Router} from '@angular/router';
+import {FormArray, FormBuilder, FormControl, FormGroup, Validators} from '@angular/forms';
+import {Recipe} from '../recipe';
+import {RecipeConverter} from '../recipe-converter';
+import {User} from '../user';
+import {Converter} from '../converter';
+
+@Component({
+  selector: 'app-branching-recipe-form',
+  templateUrl: './branching-recipe-form.component.html',
+  styleUrls: ['./branching-recipe-form.component.scss']
+})
+export class BranchingRecipeFormComponent implements OnInit {
+  baseRecipe!: Recipe;
+  baseUploaderUid!: string;
+  id: string | null = this.route.snapshot.paramMap.get('id');
+  changed: boolean = true;
+
+  constructor(
+    private fb: FormBuilder,
+    private db: AngularFirestore,
+    private route: ActivatedRoute,
+    private fAuth: AngularFireAuth,
+    private router: Router,
+  ) { 
+    this.setRecipeData().then(() => {
+      // console.log(this.baseRecipe);
+      this.basicsFormGroup.controls.name.setValue(this.baseRecipe.recipeName);
+      this.basicsFormGroup.controls.difficulty.setValue(this.baseRecipe.difficulty);
+      this.basicsFormGroup.controls.description.setValue(this.baseRecipe.description);
```

Comment:
Since it is possible you don't set baseRecipe inside setRecipeData, then it is possible here that you are accessing a null variable. I would have checks here.